remote/bitbucketserver/client.go

```diff
@@ -0,0 +1,62 @@
+package bitbucketserver
+
+import (
+	"net/http"
+	"crypto/tls"
+	log "github.com/Sirupsen/logrus"
+	"io/ioutil"
+	"encoding/pem"
+	"crypto/x509"
+	"github.com/mrjones/oauth"
+)
+
+
+func NewClient(ConsumerKey string, URL string) *oauth.Consumer{
+	//TODO: make this configurable
+	privateKeyFileContents, err := ioutil.ReadFile("/var/lib/bitbucketserver/private_key.pem")
+	log.Info("Tried to read the key")
+	if err != nil {
+		log.Error(err)
+	}
+
+	block, _ := pem.Decode([]byte(privateKeyFileContents))
+	privateKey, err := x509.ParsePKCS1PrivateKey(block.Bytes)
+	if err != nil {
+		log.Error(err)
+	}
+
```

Comment:
all of this above code should be in the `Load` section otherwise we'll end up reading the file from disk and parsing the pem on every API call which has performance implications. Instead, let's pass the privateKey into this function